linkerd/cache/src/layer.rs

```diff
@@ -27,7 +27,7 @@ impl<L: Clone, T> Clone for CacheLayer<T, L> {
 
 impl<T, L, N> tower::layer::Layer<N> for CacheLayer<T, L>
 where
-    T: Eq + std::hash::Hash,
+    T: Eq + std::hash::Hash + Send + 'static,
```

Comment:
Does the cache actually require this? Or from debugging?

Reply:
I believe this is necessary due to how the `Lock` layer works now — the type inside the lock must be `'static` so that it can go in a boxed future.